cdap-operational-stats-core/src/main/java/co/cask/cdap/operations/hdfs/HDFSInfo.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.operations.hdfs;
+
+import co.cask.cdap.operations.OperationalStats;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.Iterables;
+import org.apache.hadoop.ha.HAServiceProtocol;
+import org.apache.hadoop.ha.HAServiceStatus;
+import org.apache.hadoop.ha.HAServiceTarget;
+import org.apache.hadoop.hdfs.DFSConfigKeys;
+import org.apache.hadoop.hdfs.DFSUtil;
+import org.apache.hadoop.hdfs.DistributedFileSystem;
+import org.apache.hadoop.hdfs.HAUtil;
+import org.apache.hadoop.hdfs.tools.NNHAServiceTarget;
+import org.apache.hadoop.net.NetUtils;
+import org.apache.hadoop.util.VersionInfo;
+
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URL;
+import java.util.Collection;
+import javax.annotation.Nullable;
+
+/**
+ * {@link OperationalStats} representing HDFS information.
+ */
+@SuppressWarnings("unused")
+public class HDFSInfo extends AbstractHDFSStats implements HDFSInfoMXBean {
+  @VisibleForTesting
+  static final String STAT_TYPE = "info";
+
+  @Override
+  public String getStatType() {
+    return STAT_TYPE;
+  }
+
+  @Override
+  public String getVersion() {
+    return VersionInfo.getVersion();
+  }
+
+  @Override
+  public String getWebURL() throws IOException {
+    if (HAUtil.isHAEnabled(conf, getNameService())) {
+      return getHAWebURL().toString();
+    }
+    try (DistributedFileSystem dfs = createDFS()) {
+      return rpcToHttpAddress(dfs.getUri()).toString();
+    }
+  }
+
+  @Override
+  public String getLogsURL() throws IOException {
+    return getWebURL() + "/logs";
+  }
+
+  @Override
+  public void collect() throws IOException {
+    // TODO: no need to refresh, but need to figure out a way to not spawn a new thread for such stats
+  }
+
+  @Nullable
+  private String getNameService() {
+    Collection<String> nameservices = conf.getTrimmedStringCollection(DFSConfigKeys.DFS_NAMESERVICES);
+    if (nameservices.isEmpty()) {
+      return null;
```

Comment:
It can be null when there is no federated HDFS, which also means no HA. If it is present, it should never be empty, because ``getTrimmedStringCollection`` will return a collection with at least one element. Fixed.

However, in the first case (when nameservices is null), it is better to return null, because some users of this API (e.g. ``HAUtil.isHAEnabled()``, expect this to return null when federation is not enabled).